src/osd/PG.cc

```diff
@@ -3752,11 +3753,18 @@ int PG::build_scrub_map_chunk(
     return ret;
   }
 
-
-  get_pgbackend()->be_scan_list(map, ls, deep, seed, handle);
+  get_pgbackend()->be_scan_list(map, ls, seed, handle);
   _scan_rollback_obs(rollback_obs, handle);
   _scan_snaps(map);
 
+  if (deep) {
+    // XXX: Drop the PG lock because we are just reading the data and 
```

Comment:
Yeah, but it still sits on the thread for the entire duration.  Commonly, the OpWQ only has 1 thread/shard, so even if you drop the lock, no IO can happen.  I really think it would be better to do a bounded amount of work per work queue item.